### PR TITLE
Unit test performance evaluation and improvement

### PR DESCRIPTION
**Pull request by NaterGator**

### CODE REVIEW

**Comment by NaterGator** [comment]
See #3641

File: src/FirmwarePlugin/FirmwarePlugin.h [hunk 1]
```diff
@@ -147,7 +147,7 @@ class FirmwarePlugin : public QObject
     virtual bool sendHomePositionToVehicle(void);
     
     /// Returns the parameter that is used to identify the default component
-    virtual QString getDefaultComponentIdParam(void) const { return QString(); }
+    virtual QString getDefaultComponentIdParam(void) const { return QString("SYS_AUTOSTART"); }
```
Comment (DonLakeFlyer) [inline src/FirmwarePlugin/FirmwarePlugin.h]: This is wrong. The generic case knows nothing about the parameter set it is dealing with.

**Comment by DonLakeFlyer** [comment]
I'd rather see the threading changes and deleteLater changes in a separate pull. I'm not so sure I want that for 3.0. That whole space has been so flaky I don't want to introduce any new problems. Current code may not be quite right, but I'm pretty sure it works ok (we may leak a link, but I can live with that). Then in 3.1 we can revisit the whole threading model, deleteLater shutdown stuff.

Does that make sense?

**Comment by NaterGator** [comment]
Sure, I'm not in a position to fix this PR at the moment but I should be able to do it tonight.

**Comment by DonLakeFlyer** [comment]
Decided to shut things down for 3.0. So this will go to 3.1

**Comment by DonLakeFlyer** [comment]
@NaterGator Can you rebase this and I'll take a look at it again

**Comment by dagar** [comment]
Closing as stale.

**Pull request closed without merging**